src/Auth/BasicAuthenticate.php

```diff
@@ -75,6 +76,11 @@ public function getUser(ServerRequest $request)
     {
         $username = $request->env('PHP_AUTH_USER');
         $pass = $request->env('PHP_AUTH_PW');
+        $encoding = Configure::read('App.encoding');
+        if ($encoding && in_array($encoding, mb_list_encodings()) && $encoding != 'ISO-8859-1') {
+            $username = mb_convert_encoding($username, $encoding, 'ISO-8859-1');
+            $pass = mb_convert_encoding($pass, $encoding, 'ISO-8859-1');
```

Comment:
Ideally that string is a constant in that class, so we don't have to use a magic string 3 times (with the chance of typos).